server/util/api-utils.ts

```diff
@@ -1,9 +1,9 @@
 import { ApiRequest } from '../api';
 import { BadRequestError } from './error-types';
 
-export function assertRequestQuery<TQuery extends object>(
+export function assertRequestQuery<TQuery, TQueryKey extends keyof TQuery>(
```

Comment:
Nice improvement!

Reply:
Not sure why, but for some reason type inference is actually stronger when writing templates this way (where the key is defined in the template types). I don't know if it made any difference in this case, but it seems to be the preferred method.